src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -245,11 +270,37 @@ private SafeCurlHandle CreateRequestHandle(RequestCompletionSource state)
             }
 
             SetProxyOptions(state, requestHandle);
+            SetRequestHandleDecompressionOptions(requestHandle);
             // TODO: Handle headers and other options
 
             return requestHandle;
         }
 
+        private void SetRequestHandleDecompressionOptions(SafeCurlHandle requestHandle)
+        {
+            var curlEncoding = new StringBuilder();
+            if (AutomaticDecompression != DecompressionMethods.None)
+            {
+                var needComma = false;
+                if ((AutomaticDecompression & DecompressionMethods.GZip) != 0)
+                {
+                    curlEncoding.Append(EncodingNameGzip);
+                    needComma = true;
+                }
+
+                if ((AutomaticDecompression & DecompressionMethods.Deflate) != 0)
+                {
+                    if (needComma)
+                    {
+                        curlEncoding.Append(',');
+                    }
+
+                    curlEncoding.Append(EncodingNameDeflate);
+                }
+                Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_ACCEPTENCODING, curlEncoding.ToString());
+            }
+        }
+
```

Comment:
FWIW, you could avoid the per-call StringBuilder here and simplify the logic by changing this to be something like:
```C#
private void SetRequestHandleDecompressionOptions(SafeCurlHandle requestHandle)
{
    bool gzip = (AutomaticDecompression & DecompressionMethods.GZip) != 0;
    bool deflate = (AutomaticDecompression & DecompressionMethods.Deflate) != 0;
    if (gzip || deflate)
    {
        string encoding = 
            gzip && deflate ? EncodingNameGzip + ", " + EncodingNameDeflate :
            gzip ? EncodingNameGzip :
            EncodingNameDeflate;
        Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_ACCEPTENCODING, encoding);
}
```